src/core/client.ts

```diff
@@ -65,9 +65,70 @@ export enum ClientEventType {
   DocumentSyncResult = 'document-sync-result',
 }
 
-export interface ClientEvent {
+/**
+ * `ClientEvent` is an event that occurs in `Client`. It can be delivered using
+ * `Client.subscribe()`.
+ *
+ * @public
+ */
+export type ClientEvent =
+  | StatusChangedEvent
+  | DocumentsChangedEvent
+  | PeersChangedEvent
+  | StreamConnectionStatusChangedEvent
+  | DocumentSyncResult;
+
+export interface AbstractClientEvent {
```

Comment:
I think `AbstractClientEvent` is like a name representing an abstract class. I've seen [firebase-js-sdk](https://github.com/firebase/firebase-js-sdk/blob/5ad7ff2ae955c297556223e6cb3ad9d4b897f664/packages/firestore/src/lite/user_data_writer.ts#L57) use a similar name in the Abstract class.
I would like to suggest `ClientEventInfo` instead of this name. This name got an idea [here](https://github.com/firebase/firebase-js-sdk/blob/5ad7ff2ae955c297556223e6cb3ad9d4b897f664/packages/auth-types/index.d.ts#L21) while investigating the interface name used by firebase-js-sdk.
How about this name?

Reply:
I missed it. The name also did not end with Event, so I modified it.